### ISSUE TITLE
发现程序中一个错误

### ISSUE THREAD

**Issue by zezisme** [JY-112553/yolov1-keras-voc#1]
在tiny-yolov1.py脚本中，第150行代码：if iou_score >0.2:判断iou值大小这里应该是小于号，即：if iou_score < 0.2:

**Comment by jcui65** [comment]
我刚刚把代码下载下来，正在进行各种尝试。我还没很理解这个代码。但是试了大于0.2和小于0.2之后，我不觉得原来的代码是个错误。

**Comment by jcui65** [comment]
我觉得原来的代码是对的，如果IOU>0.2，则说明该部分（该物体块）与当前的最大的物体块的重合度够高（大于0.2），于是，这个物体块不属于一个单独的新物体，故而将其filter_mask置零，也就是消除了一个多余的，无用的物体。

**Comment by zezisme** [comment]
> 我觉得原来的代码是对的，如果IOU>0.2，则说明该部分（该物体块）与当前的最大的物体块的重合度够高（大于0.2），于是，这个物体块不属于一个单独的新物体，故而将其filter_mask置零，也就是消除了一个多余的，无用的物体。

是的，你是对的，是我当时没理解对，不过这个github项目貌似有点问题，不知道你跑出来没有，我不管是用voc数据集还是自己的，训练很多轮后，预测结果都特别差，而且预测的结果貌似都一样，跟训练不训练没有区别，都是那几个框，完全跟没读过图像信息一样

**Comment by jcui65** [comment]
我跑了30个epoch，感觉还可以。当然，我可能就挑了几张图片。你训练了多少个epoch呢？你测试了多少图片呢？

**Comment by BryceYuZh** [comment]
我在voc数据集上训练了30epoch之后 他似乎并不能训练出想要的结果

**Comment by Scrooge20291224** [comment]
> 我在voc数据集上训练了30epoch之后 他似乎并不能训练出想要的结果

我也是，一言不合就person，trust score特别低

**Comment by Scrooge20291224** [comment]
我在VOC2007全体数据跑了120个epoch，测试时效果还是很差
![YOLOv1-000002](https://user-images.githubusercontent.com/10172485/84589743-4fadb180-ae63-11ea-9a56-b3624673468b.jpg)
![YOLOv1-000003](https://user-images.githubusercontent.com/10172485/84589746-53413880-ae63-11ea-9e84-5ef8184a7e39.jpg)
![YOLOv1-000004](https://user-images.githubusercontent.com/10172485/84589749-576d5600-ae63-11ea-850a-50e97af8f457.jpg)

**Comment by Scrooge20291224** [comment]
训练200轮以上，似乎有所改观：
![YOLOv1-000002](https://user-images.githubusercontent.com/10172485/84595429-6d424180-ae8a-11ea-8908-5b079b63c00c.jpg)
![YOLOv1-000003](https://user-images.githubusercontent.com/10172485/84595438-759a7c80-ae8a-11ea-82a9-8e037e7c14bb.jpg)
![YOLOv1-000004](https://user-images.githubusercontent.com/10172485/84595442-7a5f3080-ae8a-11ea-9b28-ca6b962d8f7c.jpg)
目前loss和val_loss仍在下降，等收敛后再看看

**Comment by Scrooge20291224** [comment]
确认模型应该是没问题 ，尝试换VGG16做backbone也能正常预测，但是太容易过拟合，加了BN也没用，可能还得换回原配的Backbone试试

**Comment by georgenardes** [comment]
I have trained for more than 50 epochs with one single image and the output box doesn't get closer to the ground truth.
I think that it should overfit the data, but it is converging to a wrong value. I put a "K.print_tensor(y_pred[0, 3, 3])" in the loss function to see what is happening and the same values are predicted in all tries. The index [0,3,3] is the grid in which the object center is contained.
Has anyone got trough this like me?
